components/script/dom/bluetoothremotegattserver.rs

```diff
@@ -101,10 +103,19 @@ impl BluetoothRemoteGATTServerMethods for BluetoothRemoteGATTServer {
     }
 
     // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattserver-getprimaryservice
-    fn GetPrimaryService(&self) -> Option<Root<BluetoothRemoteGATTService>> {
+    fn GetPrimaryService(&self, service: StringOrUnsignedLong) -> Option<Root<BluetoothRemoteGATTService>> {
+        let uuid: String = match BluetoothUUID::GetService(self.global().r(),
+                                                                   service.clone()) {
+                                    Ok(_) => BluetoothUUID::GetService(self.global().r(),
+                                                                       service.clone()).unwrap().to_string(),
+                                    Err(_) => {
+                                        println!("No UUID provided!");
```

Comment:
Maybe change the return type of the method to Fallible<Option<Root<BluetoothRemoteGATTService>>> (and add the [Throws] marking to the function in webidl), so it can return like Err(err) => return Err(err). This way we can use the error message thrown by the GetService() method.